### PR TITLE
[S2GRAPH-130]: Edge.propsWithTs data type should be changed into mutable to support setter interface exist in tp3.	

### PR DESCRIPTION
**Pull request by SteamShon**
- Change data type of Edge.propsWithTs to java.util.Map.
- Make Vertex/Edge/Graph to implement Tinkerpop3 structure interface.

Note that this PR based on #99 PR.

### CODE REVIEW

File: s2core/src/main/scala/org/apache/s2graph/core/Edge.scala [hunk 1]
```diff
@@ -381,6 +645,23 @@ object Edge {
...
+      val lastDeletedAt = props.get(LabelMeta.lastDeletedAt.name).ts
+      props.remove(LabelMeta.lastDeletedAt.name)
+//      val propsWithoutLastDeletedAt = props
+//
+//      propsWithoutLastDeletedAt.forall { case (_, v) => v.ts <= lastDeletedAt }
+      var ret = true
+      val iter = props.entrySet().iterator()
+      while (iter.hasNext) {
+        val e = iter.next()
+        if (e.getValue.ts > lastDeletedAt) ret = false
```
Comment (daewon) [inline s2core/src/main/scala/org/apache/s2graph/core/Edge.scala]: The while statement does not terminate the loop even if the condition '(e.getValue.ts> lastDeletedAt)' is satisfied.

**Comment by daewon** [comment]
When you add a field, you should not forget to modify the function such as 'equals', 'hashcode', etc.

It would be nice if a test for this would be added.

I'm a little sorry to lose the advantage of the 'case class' provided by scala, but I think there is more to be gained when you support 'tinkerpop' as a native.

+1

**Comment by HyunsungJo** [comment]
* Why not remove all the code lines that were commented out?
* Will the removed tests be replaced in the future? Some of them seem to be rather critical test cases (EdgeTest, IndexEdgeTest, etc.)

File: s2core/src/main/scala/org/apache/s2graph/core/Edge.scala (hunk 1 above)
Comment (SteamShon) [inline s2core/src/main/scala/org/apache/s2graph/core/Edge.scala]: I guess you are suggesting to break when we first meet (e.getValue.ts > lastDeletedAt) condition. then I will fix on this part to break early on while loop.

**Comment by SteamShon** [comment]
@daewon actually adding `equals`, `hashCode`, `toString` for Edge/Vertex/S2Property/S2VertexProperty class reside on different PR #101. 

@HyunsungJo I will remove commented out codes. About EdgeTest, IndexEdgeTest, I think #101 has necessary code for them(just the methods @daewon  mentioned above). 

Sorry for splitting up same logical commit into different PR(I found out this while I was working on #101 ^^;;). Maybe I will fix up removed test cases(EdgeTest, IndexEdgeTest, etc) on #101.

File: s2core/src/main/scala/org/apache/s2graph/core/Edge.scala (hunk 1 above)
Comment (SteamShon) [inline s2core/src/main/scala/org/apache/s2graph/core/Edge.scala]: @daewon I just fixed it. can you review this again?

**Pull request merged**